In [8]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [9]:
!ls

data_processing.ipynb  processed_data_3_majority.csv  processed_data.csv
datasets	       processed_data_5_majority.csv


In [10]:
%cd drive/Shareddrives/CS704_Project/data_processing

[Errno 2] No such file or directory: 'drive/Shareddrives/CS704_Project/data_processing'
/content/drive/Shareddrives/CS704_Project/data_processing


In [11]:
import pandas as pd
df = pd.read_csv("processed_data_3_majority.csv")
df.head()

<ipython-input-11-8060f93470f0>:2: DtypeWarning: Columns (0,1,2,3,4,5,11,12,64,65) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("processed_data_3_majority.csv")


tweet target  \
0  @POTUS Biden Blunders - 6 Month Update\n\nInfl...   True   
1  @S0SickRick @Stairmaster_ @6d6f636869 Not as m...   True   
2  THE SUPREME COURT is siding with super rich pr...   True   
3  @POTUS Biden Blunders\n\nBroken campaign promi...   True   
4  @OhComfy I agree. The confluence of events rig...   True   

                  manual_keywords  \
0  Americans, eviction moratorium   
1  Americans, eviction moratorium   
2  Americans, eviction moratorium   
3  Americans, eviction moratorium   
4  Americans, eviction moratorium   

                                           statement 3_label_majority_answer  \
0  End of eviction moratorium means millions of A...                   Agree   
1  End of eviction moratorium means millions of A...                   Agree   
2  End of eviction moratorium means millions of A...                   Agree   
3  End of eviction moratorium means millions of A...                   Agree   
4  End of eviction moratorium means millions of A...                   Agree   

  followers_count  friends_count  favourites_count  statuses_count  \
0          4262.0         3619.0           34945.0         16423.0   
1          1393.0         1621.0           31436.0         37184.0   
2             9.0           84.0             219.0          1184.0   
3          4262.0         3619.0           34945.0         16423.0   
4            70.0          166.0           15282.0          2194.0   

   listed_count  ...  dots exclamation questions  ampersand  capitals  digits  \
0          44.0  ...   5.0         0.0       1.0        0.0      33.0     3.0   
1          64.0  ...   1.0         0.0       0.0        0.0      14.0     0.0   
2           0.0  ...   0.0         0.0       0.0        0.0       3.0     0.0   
3          44.0  ...   3.0         0.0       0.0        1.0       6.0     8.0   
4           0.0  ...   3.0         0.0       1.0        0.0      11.0     3.0   

   long_word_freq  short_word_freq  \
0             5.0             19.0   
1             2.0             34.0   
2             4.0             10.0   
3             1.0             30.0   
4             2.0             19.0   

                                       cleaned_tweet  Truthfulness  
0  Biden Blunders  6 Month Update  Inflation Delt...          True  
1  Not as many people are literally starving and ...          True  
2  THE SUPREME COURT is siding with super rich pr...          True  
3  Biden Blunders  Broken campaign promises Infla...          True  
4  I agree The confluence of events right now is ...          True  

[5 rows x 66 columns]

In [12]:
df = df[df['Truthfulness'].isin([True, False])]
X = df.select_dtypes(include='number').assign(cleaned_tweet=df['cleaned_tweet'])
y = df['Truthfulness']

In [13]:
from sklearn.model_selection import train_test_split
X_train, X_temp, y_train, y_temp = train_test_split(X, y, test_size=0.3, random_state=42)

In [14]:
X_train_1 = X_train.drop('cleaned_tweet', axis=1)
y_train_1 = y_train.map({True: 1, False: 0})

X_val_1 = X_temp.drop('cleaned_tweet', axis=1)
y_val_numeric = y_temp.map({True: 1, False: 0})

In [17]:
import pandas as pd
from sklearn.ensemble import AdaBoostClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, f1_score, precision_score, recall_score



# Define parameters
n_estimators_list = list(range(100, 1001, 100))
learning_rate_list = [0.01, 0.1, 1.0]


for n_estimators in n_estimators_list:
    for learning_rate in learning_rate_list:
      # Create an AdaBoostClassifier with a decision tree as the base estimator
      clf = AdaBoostClassifier(
          n_estimators=n_estimators,
          learning_rate=learning_rate,
          random_state=42
      )

      # Fit the model on the entire dataset
      clf.fit(X_train_1, y_train_1)

      # Make predictions on the same dataset for evaluation (not recommended for real-world scenarios)
      y_pred = clf.predict(X_val_1)

      # Calculate metrics
      accuracy = accuracy_score(y_val_numeric, y_pred)
      f1 = f1_score(y_val_numeric, y_pred)
      precision = precision_score(y_val_numeric, y_pred)
      recall = recall_score(y_val_numeric, y_pred)

      print(f"Parameters: n_estimators={n_estimators}, learning_rate={learning_rate}")
      print(f"Accuracy: {accuracy:.2f}")
      print(f"F1 Score: {f1:.2f}")
      print(f"Precision: {precision:.2f}")
      print(f"Recall: {recall:.2f}")
      print("\n" + "="*50 + "\n")

Parameters: n_estimators=100, learning_rate=0.01
Accuracy: 0.58
F1 Score: 0.54
Precision: 0.61
Recall: 0.48


Parameters: n_estimators=100, learning_rate=0.1
Accuracy: 0.62
F1 Score: 0.62
Precision: 0.63
Recall: 0.61


Parameters: n_estimators=100, learning_rate=1.0
Accuracy: 0.64
F1 Score: 0.64
Precision: 0.65
Recall: 0.64


Parameters: n_estimators=200, learning_rate=0.01
Accuracy: 0.58
F1 Score: 0.54
Precision: 0.61
Recall: 0.49


Parameters: n_estimators=200, learning_rate=0.1
Accuracy: 0.63
F1 Score: 0.63
Precision: 0.64
Recall: 0.62


Parameters: n_estimators=200, learning_rate=1.0
Accuracy: 0.64
F1 Score: 0.65
Precision: 0.65
Recall: 0.64


Parameters: n_estimators=300, learning_rate=0.01
Accuracy: 0.59
F1 Score: 0.56
Precision: 0.63
Recall: 0.51


Parameters: n_estimators=300, learning_rate=0.1
Accuracy: 0.63
F1 Score: 0.64
Precision: 0.64
Recall: 0.63


Parameters: n_estimators=300, learning_rate=1.0
Accuracy: 0.64
F1 Score: 0.65
Precision: 0.65
Recall: 0.64


Parameters: n_es

Best result obtained for</br>
Parameters: n_estimators=1000, learning_rate=1.0</br>
Accuracy: 0.65</br>
F1 Score: 0.65</br>
Precision: 0.66</br>
Recall: 0.65